### PR TITLE
added a lint task to test workflow

### PR DESCRIPTION
**Pull request by wxtim**
Fix MyPy failure on master by only running it with a version of Python we know works.

Additionally:

* Run the linting as a separate fast run/fail test before running other tests.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -80,15 +80,17 @@ graph =
 rosa =
 tests =
     aiosmtpd
+    pytest
+    # Note: some tests also depend on cylc-rose which has to be the
+    # development version installed manually (because the latest production
+    # version is pinned to the previous rose release)
+    %(lints)s
```
Comment (oliver-sanders) [inline setup.cfg:87]: Might as well keep lint deps separate, will reduce `pip install` time.

```suggestion
```

**Review by oliver-sanders — APPROVED** [review]

File: setup.cfg [hunk 2]
```diff
@@ -80,15 +80,17 @@ graph =
...
+lints =
```
Comment (MetRonnie) [inline setup.cfg:88]: `lint`?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -37,7 +37,47 @@ defaults:
...
 jobs:
+  lint:
+    runs-on: ubuntu-latest
+    timeout-minutes: 3
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.rose_ref || github.sha }}
+          path: rose
```
Comment (MetRonnie) [inline .github/workflows/test.yml:48]:
```suggestion
```

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -37,7 +37,47 @@ defaults:
...
+        with:
+          python-version: 3.9
+
+      - name: Apt-Get Install
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y shellcheck
+
+      - name: Install Rose
+        working-directory: rose
```
Comment (MetRonnie) [inline .github/workflows/test.yml:61]:
```suggestion
```

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -37,7 +37,47 @@ defaults:
...
+
+      - name: Style
+        working-directory: rose
+        run: |
+          flake8
+          etc/bin/shellchecker
+          yarn run lint
+
+      - name: MyPy
+        working-directory: rose
```
Comment (MetRonnie) [inline .github/workflows/test.yml:74]:
```suggestion
```

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -37,7 +37,47 @@ defaults:
...
+        run: |
+          pip install ."[lints]"
+          yarn install
+
+      - name: Style
+        working-directory: rose
+        run: |
+          flake8
+          etc/bin/shellchecker
+          yarn run lint
```
Comment (MetRonnie) [inline .github/workflows/test.yml:71]:
```suggestion
      - name: Flake8
        run: flake8
        
      - name: Shellchecker
        run: etc/bin/shellchecker

      - name: ESLint
        run: yarn run lint
```

File: setup.cfg (hunk 2 above)
Comment (wxtim) [reply setup.cfg:88]:
```suggestion
lint =
```

File: .github/workflows/test.yml [hunk 7]
```diff
@@ -37,7 +37,47 @@ defaults:
...
+
+      - name: Apt-Get Install
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y shellcheck
+
+      - name: Install Rose
+        working-directory: rose
+        run: |
+          pip install ."[lints]"
```
Comment (wxtim) [inline .github/workflows/test.yml:63]:
```suggestion
          pip install ."[lint]"
```

**Review by MetRonnie — APPROVED** [review]

**Pull request merged**